aten/src/ATen/core/jit_type.h

```diff
@@ -1236,6 +1236,31 @@ struct TORCH_API ComplexType : public NumberType {
   }
 };
 
+// We need to introduce `SymIntType` to represent the `SymInt` type
+// in function schemas e.g. `aten::narrow_copy(... SymInt length)
+// JIT types are tightly integrated into Dispatcher and function
+// schema handling
+struct SymIntType;
+using SymIntTypePtr = SingletonTypePtr<SymIntType>;
+struct TORCH_API SymIntType : public Type {
+  bool equals(const Type& rhs) const override {
+    return rhs.kind() == kind();
+  }
+  std::string str() const override {
+    return "SymInt";
+  }
+  std::string annotation_str_impl(TypePrinter printer = nullptr) const override {
```

Comment:
Why introduce this type instead of SymbolicIntNode then make `SymInt` parse as `Union[SymbolicIntNode | int]`